fix Function on Kubernetes for Python V2 progamming model

kevin808 opened the pull request:
<!-- Please provide all the information below.  -->

### Issue describing the changes in this PR

Resolves python programming V2 does not work with function on kubernetes

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

kevin808 commented:
@microsoft-github-policy-service agree company="Microsoft"

kshyju commented:
@vrdmr / @gavin-aguiar Could you review this?

gavin-aguiar commented on src/Azure.Functions.Cli/Actions/KubernetesActions/KubernetesDeployAction.cs:
```diff
@@ -220,7 +226,9 @@ private async Task<TriggersPayload> GetTriggersLocalFiles()
             }
 
-            var functionsJsons = GlobalCoreToolsSettings.CurrentWorkerRuntime == WorkerRuntime.dotnetIsolated
+            var useMetadata = GlobalCoreToolsSettings.CurrentWorkerRuntime == WorkerRuntime.dotnetIsolated || IsWorkerIndexingEnabled();
```
Why are we checking for dotnet isolated?

swapnil-nagar commented on src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs:
```diff
@@ -0,0 +1,174 @@
+    {
+        var functionAppPath = Path.Combine(functionsPath, "function_app.py");
+
+        var functionAppContents = await File.ReadAllTextAsync(functionAppPath);
```
Wrap I/O operations in try-catch blocks and log errors

swapnil-nagar commented on src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs:
```diff
@@ -0,0 +1,174 @@
+                {
+                    {"type", $"httpTrigger"},
+                    {"direction", "In"},
+                    {"name", "req"},
```
Lots of repeated constants can go on top of the file for readability or a method can we created to better manage bindings for different triggerType..

swapnil-nagar commented on src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs:
```diff
@@ -0,0 +1,174 @@
+    private static List<Dictionary<string, object>> ParsePythonFunctionApp(string contents)
+    {
+        var functionMetadataList = new List<Dictionary<string, object>>();
+        var functionPattern = new Regex(@"@app\.(\w+)(?:_trigger)?\(([^)]+)\)\s+def (\w+)\(", RegexOptions.Compiled | RegexOptions.Multiline);
```
NIT: Add a comment explaining the what this regex does.

swapnil-nagar commented on src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs:
```diff
@@ -0,0 +1,174 @@
+                    {"connection", properties["connection"]},
+                    {"cardinality", "One"},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
```
Checking: should you flatten the nested dictionary with an class (or an entity)?

swapnil-nagar commented on src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs:
```diff
@@ -0,0 +1,174 @@
+                });
+                    break;
+                    // Add additional cases for other trigger types if necessary
+            }
```
If a new or unknown trigger type is encountered, it silently does nothing. This can cause hard-to-debug issues. IMHO extra logging will help here.

liliankasem commented:
Thank you for the contribution @kevin808, do you mind addressing the PR feedback? Thanks!

kevin808 commented:
> Thank you for the contribution @kevin808, do you mind addressing the PR feedback? Thanks!

Sure, I will check and address them soon. Thanks for the feedback.
